app/src/pages/inside/stepPage/modals/makeDecisionModal/executionSection/optionsSection/optionsBlock/optionsBlock.jsx

```diff
@@ -51,16 +48,8 @@ export const OptionsBlock = ({
   const historyItems = useSelector(historyItemsSelector);
   const [expanded, setOptionsState] = useState(false);
   const wrapperRef = useRef();
-  useEffect(() => {
-    const handleClickOutside = (event) => {
-      if (wrapperRef && wrapperRef.current && !wrapperRef.current.contains(event.target)) {
-        setOptionsState(false);
-      }
-    };
-    document.addEventListener('click', handleClickOutside, false);
-
-    return () => document.removeEventListener('click', handleClickOutside, false);
-  }, []);
+  const clickOutside = () => setOptionsState(false);
+  useOnClickOutside(wrapperRef, clickOutside);
```

Comment:
Can you just provide a listener callback to the dropdown component?
Seems that the `ref` can be set inside of it.